### ISSUE TITLE
No branches in non-bare repository

### ISSUE THREAD

**Issue by Governa** [go-gitea/gitea#3704]
- Gitea version (or commit ref): 05abd03
- Git version:     2.15.0
- Operating system: Ubuntu
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [x] No (Unavailable right now, but probably yes)
  - [ ] Not relevant
- Log gist:

## Description
If an user creates a repository and don't push any branches to it(just tags for example) Gitea shows an 500 error.

The problem is that it is not clear for the user why the error is being shown and what to do to solve it. Even worse is that it makes it difficult to delete the repository and try again.

**Comment by Bwko** [comment]
Is this bug present in the latest master (4c6e170ceb56a8c316e523dd44137fa02ca29735)?
I've created a new repo (without branches) and pushed a tag, but no 500.

**Comment by lunny** [comment]
After #4923 merged. When a repo has tags but not branches, 500 will be disappeared but the releases tab isn't displayed. I will open another issue to improve the situation.

**Comment by melekhine** [comment]
Have this issue with one of the repos, think it was empty. This happened after I switches docker image tag from latest to 1.7.
